natlib/svcd.c

```diff
@@ -185,3 +185,68 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+static int svcd_unsubscribe( lua_State *L )
+{
+    // check number of args	
+    if (lua_gettop(L) != 4) return luaL_error(L, "Expected (targetip, svcid, attrid, ivkid)");
+    
+    uint16_t svcid = lua_tonumber(L, 2);
+    uint16_t attrid = lua_tonumber(L, 3);
+    uint16_t ivkid = lua_tonumber(L, 4);
+    // set array to var msg
+    
+    lua_getglobal(L, "SVCD"); //index 5
+    lua_pushstring(L, "msg");
```

Comment:
I don't see why we should push a string "msg" to the stack. Although msg is the name of the array variable, it is never needed for it to be represented this way. Moreover, if you want to push the same string "msg" twice, it would be more memory efficient to save it as a constant array and push it twice. 

Untitled